src/model/reference.ml

```diff
@@ -269,19 +337,34 @@ let parse whole_reference_location s :
         )
   in
 
-  let rec label_parent { kind; identifier; location } tokens : LabelParent.t =
+  let any_page { identifier; location; _ } kind next_token tokens =
+    let path () = path [ identifier ] next_token tokens in
+    match kind with
+    | `TUnknown -> `Any_path (path ())
+    | `TModule -> `Module_path (path ())
+    | `TPage -> `Page_path (path ())
+    | _ ->
+        expected ~expect_paths:true [ "module"; "page" ] location
+        |> Error.raise_exception
+  in
+
+  let rec label_parent ({ kind; identifier; location } as token) tokens :
+      LabelParent.t =
     let kind = match_reference_kind location kind in
     match tokens with
     | [] -> (
         match kind with
         | ( `TUnknown | `TModule | `TModuleType | `TType | `TClass | `TClassType
           | `TPage ) as kind ->
             `Root (identifier, kind)
+        | `TPathComponent -> `Page_path (`TRelativePath, [ identifier ])
```

Comment:
Why is it a relative path even when `identifier` is `""`?

Reply:
Case removed, see the other comment.